Appium autostart

zapletin-dk opened the pull request.

zapletin-dk committed:
Allure added

zapletin-dk committed:
Screenshot on test failure added
Some code cleanup

zapletin-dk committed:
Run scripts added

zapletin-dk committed:
Run scripts added

zapletin-dk committed:
Appium auto runner added

zapletin-dk committed:
Scripts refactored

zapletin-dk committed:
Scripts refactored

zapletin-dk committed:
Merge branch 'master' into Appium-autostart

zapletin-dk committed:
Parallel testing added

zapletin-dk committed:
Merge remote-tracking branch 'origin/Appium-autostart' into Appium-autostart

Anassstasi commented on src/main/java/configurations/AppiumServerManager.java:
```diff
@@ -0,0 +1,42 @@
+        services.clear();
+    }
+
+
```
Please fix styling issues

Anassstasi commented on src/main/java/configurations/AppiumServerManager.java:
```diff
@@ -0,0 +1,42 @@
+    private AppiumDriverLocalService service;
+    private static final Map<String, AppiumDriverLocalService> services = new HashMap<>();
+
+    public void startServer(String os, String port) {
```
Can we select port randomly? is it necessary to pass it?

zapletin-dk replied:
Yes we can with .usingAnyFreePort(). In my opinion it's preferable to use for local testing. But if we want to scale project and run tests on remote machines or via docker it would be better to use determined ports for simpler integrations.

Anassstasi commented on src/main/java/configurations/AppiumServerManager.java:
```diff
@@ -0,0 +1,42 @@
+
+public class AppiumServerManager {
+    private AppiumDriverLocalService service;
+    private static final Map<String, AppiumDriverLocalService> services = new HashMap<>();
```
Can we use threadlocal here?

zapletin-dk replied:
Yes

Anassstasi reviewed (changes requested):
Minor comments

zapletin-dk committed:
Some minor refactor
ThreadLocal added for services
Added ability to run appium server on any free port
